invoice/src/main/java/org/killbill/billing/invoice/model/InvoiceItemCatalogBase.java

```diff
@@ -127,7 +128,11 @@ public boolean matches(final Object o) {
             return false;
         }
         final InvoiceItemCatalogBase that = (InvoiceItemCatalogBase) o;
-        if (catalogEffectiveDate != null && that.catalogEffectiveDate != null && catalogEffectiveDate.compareTo(that.catalogEffectiveDate) != 0) {
+        if (catalogEffectiveDate != null &&
+            that.catalogEffectiveDate != null &&
+            catalogEffectiveDate.compareTo(that.catalogEffectiveDate) != 0 &&
+            // https://github.com/killbill/killbill/issues/1370
+            Math.abs(Seconds.secondsBetween(catalogEffectiveDate, that.catalogEffectiveDate).getSeconds()) != 0) {
```

Comment:
This seems fine, but could be simplified I think, by concatening line 133 and 135: 

`Math.abs(Seconds.secondsBetween(catalogEffectiveDate, that.catalogEffectiveDate).getSeconds())  > 1`

Reply:
True, but I find the current version more explicit (the first statement is the actual logic, the second is a workaround for DB engines limitations).